.github/workflows/linux_x86_omnibus.yml

```diff
@@ -0,0 +1,467 @@
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+env:
+  GOPROXY: https://proxy.golang.org,direct
+jobs:
+  pre_push_checker:
+    runs-on:
+      - codebuild-aws-lc-ci-github-actions-${{ github.run_id }}-${{ github.run_attempt }}
+        image:linux-5.0
+        instance-size:small
+        buildspec-override:true
+    steps:
+      - uses: actions/checkout@v5
+      - name: Get ECR Repoistory URL
+        id: ecr
+        run: echo AWS_ECR_REPO_LINUX_X86=${AWS_ECR_REPO_LINUX_X86} >> "$GITHUB_OUTPUT"
+      - uses: ./.github/actions/codebuild-docker-run
+        name: Run Container
+        with:
+          image: ${{ steps.ecr.outputs.AWS_ECR_REPO_LINUX_X86 }}:ubuntu-20.04_clang-8x_latest
+          run: |
+            ./tests/check_objects_and_errors.sh
+            ./tests/check_cmake_configure.sh
+            go run ./tests/check_licenses.go
+            ./tests/check_generated_src.sh
+            ./tests/coding_guidelines/style.sh
+            (cd util && go run ./doc.go)
+
+  c99_cplusplus98_checker:
+    runs-on:
+      - codebuild-aws-lc-ci-github-actions-${{ github.run_id }}-${{ github.run_attempt }}
+        image:linux-5.0
+        instance-size:small
+        buildspec-override:true
+    steps:
+      - uses: actions/checkout@v5
+      - name: Get ECR Repoistory URL
+        id: ecr
+        run: echo AWS_ECR_REPO_LINUX_X86=${AWS_ECR_REPO_LINUX_X86} >> "$GITHUB_OUTPUT"
+      - uses: ./.github/actions/codebuild-docker-run
+        name: Run Container
+        with:
+          image: ${{ steps.ecr.outputs.AWS_ECR_REPO_LINUX_X86 }}:ubuntu-22.04_gcc-12x_latest
+          run: ./tests/coding_guidelines/c99_cplusplus98_test.sh
+
+  posix_tests:
+    name: posix-tests-${{ matrix.image }}-${{ matrix.build32 && 'x86' || 'x86_64' }}
+    runs-on:
+      - codebuild-aws-lc-ci-github-actions-${{ github.run_id }}-${{ github.run_attempt }}
+        image:linux-5.0
+        instance-size:large
+        buildspec-override:true
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - amazonlinux-2_clang-7x_latest
+          - amazonlinux-2_gcc-7x_latest
+          - amazonlinux-2023_clang-15x_latest
+          - amazonlinux-2023_gcc-11x_latest
+          - centos-7_gcc-4x_latest
+          - centos-8_gcc-8x_latest
+          - ubuntu-18.04_clang-6x_latest
+          - ubuntu-18.04_gcc-7x_latest
+          - ubuntu-20.04_clang-7x_latest
+          - ubuntu-20.04_clang-8x_latest
+          - ubuntu-20.04_clang-9x_latest
+          - ubuntu-20.04_clang-10x_latest
+          - ubuntu-20.04_gcc-7x_latest
+          - ubuntu-20.04_gcc-8x_latest
+          - ubuntu-22.04_gcc-10x_latest
+          - ubuntu-22.04_gcc-11x_latest
+          - ubuntu-22.04_gcc-12x_latest
+        build32: [0] 
+        include:
+          - image: centos-7_gcc-4x_latest
+            build32: 1
+          - image: fedora-31_clang-9x_latest
+            cc: clang-9
+            cxx: clang++-9
+            build32: 0
```

Comment:
NP:

```
build32: [0]
```

Is this line needed? I suppose the line `${{ matrix.build32 || 0 }}` expression would fail otherwise.
* I ask b/c I noticed that you're able to reference `matrix.cc` below even though it's not listed as a dimension for the matrix.

Reply:
You can reference variables not in the matrix technically. But not defining this in the top-level creates the issue I described on the call where it was not creating the expected x86 and x86-64 test jobs for centos and ubuntu, but was instead collapsing it. So really the defaulting behavior is no longer needed. Can cleanup further later.